src/mod/irc.mod/chan.c

```diff
@@ -1335,6 +1335,60 @@ static int got354(char *from, char *msg)
   return 0;
 }
 
+/* React to IRCv3 CHGHOST command. CHGHOST changes the hostname and/or
+ * ident of the user. Format:
+ * :geo!awesome@eggdrop.com CHGHOST tehgeo foo.io
+ * changes user hostmask to tehgeo@foo.io
+ */
+static int gotchghost(char *from, char *msg){
+  struct flag_record fr = { FR_GLOBAL | FR_CHAN, 0, 0, 0, 0, 0 };
+  struct userrec *u;
+  struct chanset_t *chan;
+  memberlist *m;
+  char mask[1024], *nick, *ident, buf[MSGMAX], *s1=buf, *chname;
+
+  strlcpy(s1, from, sizeof buf);
+  nick = splitnick(&s1);
+  ident = newsplit(&msg);  /* Get the ident */
+  /* Update my own internal hostmask */
+  if (match_my_nick(nick)) {
+    strlcpy(botuserhost, from, sizeof botuserhost);
+  }
+  u = get_user_by_host(from);
+  /* Run the bind for each channel the user is on */
+  for (chan = chanset; chan; chan = chan->next) {
+    chname = chan->dname;
+    m = ismember(chan, nick);
+    if (m) {
+      snprintf(m->userhost, sizeof m->userhost, "%s@%s", ident, msg);
+      snprintf(mask, sizeof mask, "%s %s", chname, from);
+      check_tcl_chghost(nick, from, mask, u, chname, ident, msg);
+      get_user_flagrec(m->user ? m->user : get_user_by_host(s1), &fr,
```

Comment:
why is the flag getting after calling the bind, shouldn't it be before so it filters?

Reply:
Pulling the flag record for the bind triggering is currently done inside check_tcl_chghost. But to your point, the flagrec is then pulled again here to be passed to check_this_member, which rechecks what status flags the user should receive on IRC, if any. I guess we could consolidate and call it before check_tcl_chghost, and then pass it to the function?